add-on/src/lib/ipfs-proxy/access-control.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+/* eslint-env browser */
+
+const EventEmitter = require('events')
+const PQueue = require('p-queue')
+
+class AccessControl extends EventEmitter {
+  constructor (storage, key = 'ipfsProxyAcl') {
+    super()
+    this._storage = storage
+    this._key = key
```

Comment:
What if we store ACL keys with a prefix?
eg.  set `this._key = 'ipfsProxyAcl.' + key` (or add/remove a prefix in `getAcl` and `_setAcl`)
It removes risk of overriding storage keys + makes it easier to browse/sort local storage keys when debugging the extension.

Reply:
So, just to confirm our keys would look like `'ipfsProxyAcl.http://google.co.uk` for example?

Currently things look sort of like this:

```js
{
  ipfsProxyAcl: {
    'http://google.co.uk': {
      'files.add': true
    },
    'https://ipfs.io': {
      'files.add': false,
      'object.new': false
    }
  }
}
```

...and we'd be altering it to be something like:

```js
{
  'ipfsProxyAcl.http://google.co.uk': {
    'files.add': true
  },
  'ipfsProxyAcl.https://ipfs.io': {
    'files.add': false,
    'object.new': false
  }
}
```

I like this and I think it makes things simpler, the only issue I can think of is getting the entire ACL for the manage permissions page might be tricky - we'd have to read everything in storage into memory and filter just the keys we need.

I think that's a better trade of though since atm we're reading the entire ACL into memory for every `getAccess` and `setAccess`, which will be far more commonly called than `getAcl`.

Reply:
Sounds good. 
I think its the only thing that blocks merging this  PR, other things can be addressed in separate PRs